src/components/modes/student/StudentButtons.js

```diff
@@ -196,11 +203,17 @@ class StudentButtons extends Component {
   }
 }
 
-const mapStateToProps = ({ context, appInstanceResources, code }) => {
+const mapStateToProps = ({
+  context,
+  appInstanceResources,
+  code,
+  appInstance,
+}) => {
   const { userId, offline, appInstanceId } = context;
   const inputResource = appInstanceResources.content.find(({ user, type }) => {
     return user === userId && type === INPUT;
   });
+  const { visibility } = appInstance.content.settings;
```

Comment:
I think this needs to fall back here to `DEFAULT_VISIBILITY` so that it's backwards compatible with settings where this key does not exist. Otherwise, it might just be `undefined`.